[Win WebKitTestRunner] TestRunner::setOpenPanelFiles should accept a Windows absolute file path

fujii opened the pull request:
#### 2bbf0fb389fb383fd17422ac26a047d705cda179
<pre>
[Win WebKitTestRunner] TestRunner::setOpenPanelFiles should accept a Windows absolute file path
<a href="https://bugs.webkit.org/show_bug.cgi?id=277772">https://bugs.webkit.org/show_bug.cgi?id=277772</a>

Reviewed by Ross Kirsling.

Windows port failed some layout tests which create a temporary file
and passes the absolute path to testRunner.setOpenPanelFiles.
<a href="https://commits.webkit.org/220142@main">https://commits.webkit.org/220142@main</a> added the absolute path support
of setOpenPanelFiles by checking a leading forward slash character for
Unix.

Use PathIsRelativeA and UrlCreateFromPathA API for Windows.

Additionally, a temporarily created WKURLRef object of &quot;file://&quot; was
leaked, Use WKRetainPtr&lt;WKURLRef&gt; and adoptWK for it.

* LayoutTests/platform/win/TestExpectations:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::makeOpenPanelURL):
(WTR::TestRunner::setOpenPanelFiles):

Canonical link: <a href="https://commits.webkit.org/282032@main">https://commits.webkit.org/282032@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e1eaf41170d33c2104c1a1e59e77d6c7dfc554ed

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61839 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41193 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14431 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/65818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/12384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63958 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48879 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12655 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/65818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/12384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/64908 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/38255 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/53558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/65818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/34899 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/10791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/11315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/56708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/11095 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/67547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5782 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/10830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/67547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5807 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/53509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/67547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/88/builds/4740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9312 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/36993 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/38077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/39173 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/37822 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
```diff
@@ -1798,6 +1803,25 @@ void TestRunner::setMockGamepadButtonValue(unsigned, unsigned, double)
+#else
+    WKRetainPtr<WKURLRef> fileURL;
+    if (filePath[0] == '/') {
+        fileURL = adoptWK(WKURLCreateWithUTF8CString("file://"));
+        baseURL = fileURL.get();
```
How come we're changing this from just `baseURL = WKURLCreateWithUTF8CString("file://");`?

fujii replied:
It was leaked. I will add a comment in the commit log.

fujii committed:
[Win WebKitTestRunner] TestRunner::setOpenPanelFiles should accept a Windows absolute file path
https://bugs.webkit.org/show_bug.cgi?id=277772

Reviewed by Ross Kirsling.

Windows port failed some layout tests which create a temporary file
and passes the absolute path to testRunner.setOpenPanelFiles.
https://commits.webkit.org/220142@main added the absolute path support
of setOpenPanelFiles by checking a leading forward slash character for
Unix.

Use PathIsRelativeA and UrlCreateFromPathA API for Windows.

Additionally, a temporarily created WKURLRef object of "file://" was
leaked, Use WKRetainPtr<WKURLRef> and adoptWK for it.

* LayoutTests/platform/win/TestExpectations:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::makeOpenPanelURL):
(WTR::TestRunner::setOpenPanelFiles):

Canonical link: https://commits.webkit.org/282032@main
